chain/ethereum/src/ethereum_adapter.rs

```diff
@@ -1588,14 +1588,21 @@ pub(crate) fn parse_log_triggers(
 pub(crate) fn parse_call_triggers(
     call_filter: &EthereumCallFilter,
     block: &EthereumBlockWithCalls,
-) -> Vec<EthereumTrigger> {
+) -> anyhow::Result<Vec<EthereumTrigger>> {
     match &block.calls {
         Some(calls) => calls
             .iter()
             .filter(move |call| call_filter.matches(call))
-            .map(move |call| EthereumTrigger::Call(Arc::new(call.clone())))
+            .map(
+                move |call| match block.transaction_for_call_succeeded(call) {
+                    Ok(true) => Ok(Some(EthereumTrigger::Call(Arc::new(call.clone())))),
+                    Ok(false) => Ok(None),
+                    Err(e) => Err(e),
+                },
+            )
+            .filter_map_ok(|some_trigger| some_trigger)
```

Comment:
This ended up a bit convoluted, but the point is that we need to keep `Err`s  and drop `Ok(None)`s for the conversion to `Result<Vec<_>>` to work.

Reply:
At this point I'd just use a `for` so that you can use `?`, but this also works.